src/os_posix/os_union_fs.c

```diff
@@ -891,27 +883,38 @@ __union_build_extents_from_dest_file_lseek(
 
     /* Check that we opened a valid file descriptor. */
     WT_ASSERT(session, fcntl(fd, F_GETFD) != -1 || errno != EBADF);
-    __union_fs_file_size((WT_FILE_HANDLE *)union_fh, (WT_SESSION *)session, &file_size);
+    WT_ERR(__union_fs_file_size((WT_FILE_HANDLE *)union_fh, (WT_SESSION *)session, &file_size));
     __wt_verbose_debug2(session, WT_VERB_FILEOPS, "File: %s", filename);
     __wt_verbose_debug2(session, WT_VERB_FILEOPS, "    len: %ld", file_size);
 
-    // Find the next data block. data_end_offset is initialized to zero so we start from the
-    // beginning of the file.
+    if (file_size > 0) {
+        /* Initialize the hole_list as one big hole. Then find data segments and remove them. */
+        WT_ERR(__wt_calloc_one(session, &union_fh->destination.hole_list));
+        union_fh->destination.hole_list->off = 0;
+        union_fh->destination.hole_list->len = (size_t)file_size;
+        union_fh->destination.hole_list->next = NULL;
+    }
+
+    /*
+     * Find the next data block. data_end_offset is initialized to zero so we start from the
```

Comment:
Can you leave a comment here that elaborates on the lseek logic a bit: "Seeking data from offset will return offset if offset already points to data. In other words when we start at offset 0, if there is data at that offset we won't unintentionally skip it."

Reply:
Added in [9f8e50e](https://github.com/wiredtiger/wiredtiger/pull/11268/commits/9f8e50e9e9e25c96dd296661234032fd94c5c45c)